src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_BasePatternSwitchLocalRewriter.cs

```diff
@@ -351,6 +358,7 @@ private void LowerDecisionDagCore(BoundDecisionDag decisionDag)
                         !loweredNodes.Contains(nodesToLower[indexOfNode + 2]) ? nodesToLower[indexOfNode + 2] : null;
 
                     _loweredDecisionDag.Add(_factory.ExpressionStatement(sideEffect));
+                    _loweredDecisionDag.Add(_factory.HiddenSequencePoint());
```

Comment:
Is this a case where we may be executing user code through a conversion? If so could you add a comment that that is why we need a sequence point here?

Reply:
Not through a conversion, but through a `Deconstruct`, `ITuple` method, or property fetch. I'll add a comment.

---
In reply to: [184817826](https://github.com/dotnet/roslyn/pull/26138#discussion_r184817826) [](ancestors = 184817826)

Reply:
Oh, no, here it is just a built-in conversion. The sequence point is not necessary here.

---
In reply to: [184823087](https://github.com/dotnet/roslyn/pull/26138#discussion_r184823087) [](ancestors = 184823087,184817826)